Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -523,14 +523,21 @@ static inline void continueAfterRequest(RetainPtr<id <ASCCredentialProtocol>> cr
     }).get()];
 }
 
-void WebAuthenticatorCoordinatorProxy::isConditionalMediationAvailable(QueryCompletionHandler&& handler)
+static inline bool isConditionalMediationAvailableImpl(const WebCore::SecurityOriginData& data)
 {
-    handler([getASCWebKitSPISupportClass() shouldUseAlternateCredentialStore]);
+    if ([getASCWebKitSPISupportClass() respondsToSelector:@selector(canCurrentProcessAccessPasskeysForRelyingParty:)])
+        return [getASCWebKitSPISupportClass() canCurrentProcessAccessPasskeysForRelyingParty:data.securityOrigin()->domain()] && [getASCWebKitSPISupportClass() shouldUseAlternateCredentialStore];
+    return false;
+}
+
+void WebAuthenticatorCoordinatorProxy::isConditionalMediationAvailable(const WebCore::SecurityOriginData& data, QueryCompletionHandler&& handler)
+{
+    handler(isConditionalMediationAvailableImpl(data));
 }
 
 void WebAuthenticatorCoordinatorProxy::isUserVerifyingPlatformAuthenticatorAvailable(const SecurityOriginData& data, QueryCompletionHandler&& handler)
 {
-    if ([getASCWebKitSPISupportClass() shouldUseAlternateCredentialStore]) {
+    if (isConditionalMediationAvailableImpl(data)) {
```

Comment:
If this returns false, is there any chance the user will be able to use the platform authenticator. It's possible we should either remove or modify the local service call below.

Reply:
I've done some restructuring to the methods in this file:

Helper (now named canCurrentProcessAccessPasskeysForRelyingParty per suggestion) now only checks if the SPI support class responds to the new SPI message and calls it.

isConditionalMediationAvailable
1. Call helper to see if passkey access is allowed, call handler with false if not
2. Call handler with the result of shouldUseAlternateCredentialStore

isUserVerifyingPlatformAuthenticatorAvailable
1. Call helper to see if passkey access is allowed, call handler with false if not
2. If alternate credential store is available, do the existing check of disallowed relying parties
3. Otherwise return LocalService::isAvailable()

I think this is the right way to add the new check to existing flows.